### PR TITLE
nfq: Ensure packet release function set

### PR DESCRIPTION
**Pull request by jlucovsky**
This PR ensures that early error returns in `NFQCallBack` will have the packet's release function pointer set.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [5916](https://redmine.openinfosecfoundation.org/issues/5916)

Describe changes:
- Set packet's `ReleasePacket` earlier for early error paths.

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 12784

File: src/source-nfq.c [hunk 1]
```diff
@@ -532,6 +532,9 @@ static int NFQCallBack(struct nfq_q_handle *qh, struct nfgenmsg *nfmsg,
     if (nfq_config.bypass_mask) {
         p->BypassPacketsFlow = NFQBypassCallback;
     }
+
+    p->ReleasePacket = NFQReleasePacket;
```
Comment (victorjulien) [inline src/source-nfq.c:536]: I feel this belongs in `NFQSetupPkt` instead.

Comment (jlucovsky) [reply src/source-nfq.c:536]: Will do.

**Comment by jlucovsky** [comment]
Continued in #8630

**Pull request closed without merging**